File: q1.py
a=[]
b=[]

with open('input.txt', 'r') as file:
        data = file.read()
        for line in data.split('\n'):
            # print(line.split("   ")[1])  
            a.append(int(line.split("   ")[0]))
            b.append(int(line.split("   ")[1]))
a.sort()
b.sort()
sum=0

for i in range(len(a)): 
    sum =sum+  abs(a[i]-b[i])
print(sum)

count_dict = {}

for i in range(len(a)): 
    if a[i] in b:
        count_dict[a[i]] = b.count(a[i])
print(count_dict)

total_dict=0

for i in range(0,len(a)):
    if a[i] in count_dict.keys():
        total_dict=total_dict+count_dict[a[i]]*a[i]

print()
print(total_dict)

File: q1.cpp
#include <bits/stdc++.h>

using namespace std;

int main()
{
    ifstream inputFile("input.txt");
    if (!inputFile)
    {
        cerr << "Unable to open file input.txt";
        return 1;
    }
    vector<long> a, b;
    string line;
    while (getline(inputFile, line))
    {
        stringstream ss(line);
        long num;
        while (ss >> num)
        {
            a.push_back(num);
            cout << num << endl;
        }
    }
    sort(a.begin(), a.end());
    sort(b.begin(), b.end());
    long sum = 0;
    int n = a.size();
    for (int i = 0; i < n; i++)
    {
        cout << a[i] << " " << b[i] << endl;
        sum += abs(a[i] - b[i]);
    }
    cout << sum << endl;
    inputFile.close();

    return 0;
}